tests/e2e/lib/repoutils

```diff
@@ -76,3 +76,21 @@ install_qm_rpms() {
 
   dnf install -y bluechi-ctl bluechi-agent bluechi-controller qm hostname
 }
+
+fix_qm_rlimits() {
+  ###########################################################################
+  # Description:                                                            #
+  # Fix: Issue #666 need to set QM rlimit in containers.conf                #
+  #                                                                         #
+  # Arguments: None                                                         #
+  ###########################################################################
+    local ulimit_nofile
+    local ulimit_nproc
+    info_message "fix_qm_rlimits(): prepare qm containers.conf file, update ulimits"
+    exec_cmd "test -e ${QM_CTR_CFG}"
+    ulimit_nofile=$(printf %.0f "$(echo "$(ulimit -n) * $U_NOFILE_PRCTG" | bc)")
+    ulimit_nproc=$(printf %.0f "$(echo "$(ulimit -u) * $U_NPROC_PRCTG" | bc)")
+    exec_cmd "sed -i -E 's/(default_ulimits = \[)/\1\"nproc=$ulimit_nproc:$ulimit_nproc\",\"nofile=$ulimit_nofile:$ulimit_nofile\"/' $QM_CTR_CFG"
+    info_message "fix_qm_rlimits(): verify limits are set"
+    exec_cmd "grep -o '\(nproc.*\",\|nofile.*\"\)' ${QM_CTR_CFG} | sed 's/\"\|,//g'"
+}
```

Comment:
How about:

set_qm_rlimits() {
    info_message "Setting QM rlimits(): Preparing QM containers.conf file and updating ulimits"

    # Check if QM containers.conf exists
    exec_cmd "test -e ${QM_CTR_CFG}"

    # Calculate ulimits
    local ulimit_nofile=$(( $(ulimit -n) * U_NOFILE_PRCTG / 100 ))
    local ulimit_nproc=$(( $(ulimit -u) * U_NPROC_PRCTG / 100 ))

    # Update containers.conf with new limits
    exec_cmd "sed -i -E 's/(default_ulimits = \[)/\1\"nproc=${ulimit_nproc}:${ulimit_nproc}\",\"nofile=${ulimit_nofile}:${ulimit_nofile}\"/' ${QM_CTR_CFG}"

    # Verify limits are set
    info_message "fix_qm_rlimits(): Verifying limits are set"
    exec_cmd "grep -o '\(nproc.*\",\|nofile.*\"\)' ${QM_CTR_CFG} | tr -d '\"' | tr -d ','"
}


few points:
 - Used shell arithmetic $(( ... )) instead of printf and bc for simplicity and better performance.
 -  tr for Cleanup: Replaced sed for removing quotes and commas with tr for simplicity and readability.
 - please give it a full test, test just locally in a local env.

Reply:
the test is running in fedora gate refer this 
[tests/qm-sanity-test/check_qm_podman_quadlet_is_ok.sh](https://github.com/containers/qm/pull/670/files#diff-8e81340563df3ad40e52cdd0501369b27e0b7467700c085a1e5c2ed598fb88a8)

In case of tr, i failed the exec-cmd with tr.